core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -569,8 +575,10 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
           Thread.sleep(9999999)
         }
         // second attempt succeeds immediately
+        SparkContextSuite.taskSucceeded = true
       }
     }
+    assert(SparkContextSuite.taskSucceeded)
```

Comment:
Both listener and the task are both setting taskSuceeded ? That does not look right ...
I am assuming we need one failure to be raised with the appropriate message, one task success - to ensure listener success.
Additionally, re-execution of task to indicate success of task (though this aspect should be covered in some other test already).